feat: apply a provided version

LBF38 opened the pull request:
introduce a new feature to allow users to provide their version in a string format (major.minor.patch) and automatically increment the build number accordingly. 
This allows for better integration w/ other tools that can automatically determine the version number

For example, it can be integrated with semantic release tools that can determine the next version number.
Then, in the CI pipeline, the version number will be used with this Fastlane plugin to update the Flutter's project `pubspec.yaml`.

Hope you'll consider it !

LBF38 committed:
feat: apply a provided version

introduce a new feature to allow users to provide their version in a string format (major.minor.patch) and automatically increment the build number accordingly.
This allows for better integration w/ other tools that can automatically determine the version number

LBF38 committed:
docs: update option description

to be clearer

M97Chahboun commented:
@LBF38 Thank you for your contribution
Can you please add [tests](https://github.com/M97Chahboun/fastlane-plugin-flutter_bump_version/blob/645e7245714c1a44c11e39da62eb0e1f7a3bb9c8/spec/flutter_bump_version_action_spec.rb) for your changes

LBF38 commented:
@M97Chahboun Thanks for approving it !
I'll add tests as soon as possible.

Btw, what is needed for the cla check to pass ?

M97Chahboun commented:
Thank you!
You need to Sign [Contributor License Agreements](https://cla.developers.google.com/clas)

LBF38 commented:
@M97Chahboun I didn't see the "rescan" button to update the check 😄 Now, it's done !

LBF38 committed:
refactor: update fastlane action w/ tests

add tests for the new feature (provided version)
update the action to pass all tests

LBF38 commented:
@M97Chahboun done !

M97Chahboun commented:
Yes, Thank you for your contribution! 🚀

LBF38 committed:
chore: bump version to 0.3.2

M97Chahboun commented:
Please push other commit for solve follow GitHub problem, but with 0.3.0

![Screenshot 2025-02-25 at 14 33 59](https://github.com/user-attachments/assets/d7b77586-bc0e-4a33-8dd1-c324f1779510)

LBF38 committed:
chore: bump version to 0.3.0

LBF38 committed:
docs: update main README

LBF38 committed:
ci: update test workflow

M97Chahboun commented:
I will merge it ASAP (we will ignore ci workflow and create new task for fix it)
